src/IonImport.ts

```diff
@@ -28,58 +28,57 @@ import { SymbolIndex } from "./IonSymbolIndex";
  * @see http://amzn.github.io/ion-docs/symbols.html#imports
  */
 export class Import {
-  private readonly _offset: number;
-  private readonly _length: number;
-  private readonly index: SymbolIndex = {};
+    private readonly _offset: number;
+    private readonly _length: number;
+    private readonly index: SymbolIndex = {};
+    private readonly _parent : Import;
+    private readonly _symbolTable : SharedSymbolTable;
 
-  constructor(
-    private readonly _parent: Import,
-    private readonly _symbolTable: SharedSymbolTable,
-    length?: number
-  ) {
-    this._offset = (_parent && (_parent.offset + _parent.length)) || 1;
-    this._length = length || _symbolTable.symbols.length;
+    constructor(parent : Import, symbolTable : SharedSymbolTable, length?: number) {
+        this._parent = parent;
+        this._symbolTable = symbolTable;
+        this._offset = this.parent ? this.parent.offset + this.parent.length : 1;//this is wrong.
```

Comment:
Comments like "this is wrong" don't inspire confidence that this is not wrong...

Reply:
sorry took me awhile to see how the exclusive/inclusive offset numbers matched up with maxID and multiple imports interacting with each other. It is correct I just wasn't convinced at first. I'll remove it.